guide/src/templates/builtin_placeholders.md

```diff
@@ -0,0 +1,35 @@
+# Builtin placeholders
+
+`cargo-generate` supports a number of builtin placeholders for use in templates.
+
+These placeholders can be used directly in files using the [Liquid language][liquid], or from
+[Rhai scripts][rhai] using the syntax: `variable::get("placeholder name")`.
+
+The current supported builtin placeholders are:
+
+* `{{authors}}`
+
+  this will be filled in by a function borrowed from Cargo's source code, that determines your information from Cargo's configuration. It will either be on the form `username <email>` or just plain `username`.
+* `{{project-name}}`
+
+  this is supplied by either passing the `--name` flag to the command or working with the interactive CLI to supply a name.
+* `{{crate_name}}`
+
+  the snake_case_version of `project-name`
+* `{{crate_type}}`
+
+  this is supplied by either passing the `--bin` or `--lib` flag to the command line, contains either `bin` or `lib`, `--bin` is the default
+* `{{os-arch}}`
+
+  contains the current operating system and architecture ex: `linux-x86_64`
+* `{{username}}`
+
+  this will be filled in by a function borrowed from Cargo's source code, that determines your information from Cargo's configuration.
+
+* `{{cargo_embedded}}`
```

Comment:
I really like the idea, but somehow the name `cargo_embedded` does not really speak to me. The terminus `embedded` is somehow very much connotated by embedded rust.

Unfortunately, I don't have a very crisp alternative to suggest.
Maybe some things that came to mind, but are probably also not good enough yet:
- `inside_a_cargo_project`
- `project_exists_already`
- `no_new_project`
- `has_existing_project`

Reply:
I agree 100% - should *not* call it anything with embedded, but that was the name that came up at the time 😄 

Of the suggestions you made, I like `inside_a_cargo_project` the most.

But there's also the plain
* `cargo_subfolder`

Another possibility is to create a `Rhai` module, maybe call it `cargo`, and create a function for the test on that. The template could still detect the situation (though it would require a scripl).

In the same direction, what it the `file` module just got a function to test if a file can be found in a parent folder? This would open up the possibility for all languages - though I admit I hate loosing the easy way that the placeholder creates for Rust. Maybe both??

Reply:
Using a new `Rhai` module also opens the possibility to create functionality to test against e.g. dependencies (is Tokio available?), or to auto add dependencies etc....

But this might just be too powerfull!

Reply:
I will let it sink in :)
But the tendency is right now a regular build in placeholder, not a rhai script.

Reply:
Same here

Reply:
Possibly, `within_cargo_project`?

Reply:
I liked the name @barafael suggested the most - shortest, to the point, and descriptive. That, and we talked at the Nürnberg meetup yesterday, so I'm a bit biased 😄

Reply:
Sounds perfect, let's roll with this name then.

Nürnberg has a rust meetup now? How cool..